### PR TITLE
[WIP] Fix warnings in test

### LINKED ISSUE #1061: Fix warnings in test
## 🚀 Feature

Currently, we have several warnings raised during the tests from our side but not catched with `pytest.warns`:
- https://travis-ci.org/github/pytorch/ignite/jobs/689725736#L3577
- https://travis-ci.org/github/pytorch/ignite/jobs/689725736#L3589
- https://travis-ci.org/github/pytorch/ignite/jobs/689725736#L3593
- etc

We need to improve the code of tests by introducing
```python
with pytest.warns(...):
    # code that raises a warning
```

### PR DESCRIPTION
**Pull request by InCogNiTo124**
Fixes #1061 

Description:
Some tests return warnings. This PR is to catch them.

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by InCogNiTo124** [comment]
I was not able to fix `test_setup_visdom_logging` from `tests/ignite/contrib/engines/test_common.py`

File: tests/ignite/contrib/engines/test_common.py [hunk 1]
```diff
@@ -376,14 +376,15 @@ def test_setup_plx_logging():
...
 
-    _test_setup_logging(
-        setup_logging_fn=setup_plx_logging,
-        kwargs_dict={},
-        output_handler_cls=handlers.polyaxon_logger.OutputHandler,
-        opt_params_handler_cls=handlers.polyaxon_logger.OptimizerParamsHandler,
-        with_eval=False,
-        with_optim=False,
-    )
+    with pytest.warns(DeprecationWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/engines/test_common.py:379]: Please, specify warn part of message to match:
```python
with pytest.warns(DeprecationWarning, match=r"bla-bla-bla"):
    # ...
```

File: tests/ignite/contrib/handlers/test_base_logger.py [hunk 2]
```diff
@@ -154,29 +153,29 @@ def update_fn(engine, batch):
...
-    _test(cpe2.Events.ITERATIONS_15_COMPLETED, nf, cpe2)
-
-    n_custom_epochs = 2
-    cpe3 = CustomPeriodicEvent(n_epochs=n_custom_epochs)
-    n = n_epochs / n_custom_epochs
-    nf = math.floor(n)
-    ns = nf + 1 if nf < n else nf
-    _test(cpe3.Events.EPOCHS_2_STARTED, ns, cpe3)
-    _test(cpe3.Events.EPOCHS_2_COMPLETED, nf, cpe3)
+        n_iterations = 10
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_base_logger.py:156]: I think all those indents are incorrect. Function `_test` is called in various confs.

File: tests/ignite/contrib/handlers/test_custom_events.py [hunk 3]
```diff
@@ -8,55 +8,58 @@
 
 def test_bad_input():
 
-    with pytest.raises(ValueError):
-        CustomPeriodicEvent(n_iterations="a")
+    with pytest.warns(DeprecationWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_custom_events.py:11]: Please, specify warn part of message to match, like above.

File: tests/ignite/contrib/handlers/test_custom_events.py [hunk 4]
```diff
@@ -8,55 +8,58 @@
...
+        with pytest.raises(ValueError):
+            CustomPeriodicEvent(n_iterations="a")
 
-    with pytest.raises(ValueError):
-        CustomPeriodicEvent(n_iterations=0)
+        with pytest.raises(ValueError):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_custom_events.py:15]: Does it work with a single `pytest.warns` to catch multiple warnings ?

File: tests/ignite/contrib/handlers/test_custom_events.py [hunk 5]
```diff
@@ -8,55 +8,58 @@
...
 
 
 def test_new_events():
     def update(*args, **kwargs):
         pass
 
-    engine = Engine(update)
-    cpe = CustomPeriodicEvent(n_iterations=5)
-    cpe.attach(engine)
+    with pytest.warns(DeprecationWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_custom_events.py:41]: Please, specify warn part of message to match, like above.

File: tests/ignite/contrib/handlers/test_custom_events.py [hunk 6]
```diff
@@ -8,55 +8,58 @@
...
+        assert hasattr(cpe.Events, "ITERATIONS_5_COMPLETED")
 
-    assert engine._allowed_events[-2] == getattr(cpe.Events, "ITERATIONS_5_STARTED")
-    assert engine._allowed_events[-1] == getattr(cpe.Events, "ITERATIONS_5_COMPLETED")
+        assert engine._allowed_events[-2] == getattr(cpe.Events, "ITERATIONS_5_STARTED")
+        assert engine._allowed_events[-1] == getattr(cpe.Events, "ITERATIONS_5_COMPLETED")
 
-    cpe = CustomPeriodicEvent(n_epochs=5)
-    cpe.attach(engine)
+    with pytest.warns(DeprecationWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_custom_events.py:53]: Same here.

File: tests/ignite/contrib/handlers/test_custom_events.py [hunk 7]
```diff
@@ -65,9 +68,9 @@ def update(*args, **kwargs):
             pass
 
         engine = Engine(update)
-
-        cpe = CustomPeriodicEvent(n_iterations=n_iterations)
-        cpe.attach(engine)
+        with pytest.warns(DeprecationWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_custom_events.py:71]: Same here

File: tests/ignite/contrib/handlers/test_custom_events.py [hunk 8]
```diff
@@ -114,8 +117,9 @@ def update(*args, **kwargs):
     engine = Engine(update)
 
     n_epochs = 3
-    cpe = CustomPeriodicEvent(n_epochs=n_epochs)
-    cpe.attach(engine)
+    with pytest.warns(DeprecationWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_custom_events.py:120]: Same here

File: tests/ignite/contrib/handlers/test_lr_finder.py [hunk 9]
```diff
@@ -217,8 +217,9 @@ def test_plot(lr_finder, to_save, dummy_engine, dataloader):
     with lr_finder.attach(dummy_engine, to_save) as trainer_with_finder:
         trainer_with_finder.run(dataloader)
 
-    lr_finder.plot()
-    lr_finder.plot(skip_end=0)
+    with pytest.warns(UserWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_lr_finder.py:220]: Same here

File: tests/ignite/contrib/handlers/test_tqdm_logger.py [hunk 10]
```diff
@@ -373,7 +373,8 @@ def test_pbar_on_custom_events(capsys):
 
     engine = Engine(update_fn)
     pbar = ProgressBar()
-    cpe = CustomPeriodicEvent(n_iterations=15)
+    with pytest.warns(DeprecationWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_tqdm_logger.py:376]: Same here.

File: tests/ignite/contrib/handlers/test_trains_logger.py [hunk 11]
```diff
@@ -582,17 +582,18 @@ def update_fn(engine, batch):
 
     trainer = Engine(update_fn)
 
-    TrainsLogger.set_bypass_mode(True)
-    logger = TrainsLogger(output_uri=dirname)
+    with pytest.warns(UserWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_trains_logger.py:585]: Here too

File: tests/ignite/contrib/handlers/test_trains_logger.py [hunk 12]
```diff
@@ -606,28 +607,29 @@ def test_integration_as_context_manager(dirname):
     def update_fn(engine, batch):
         return next(losses_iter)
 
-    TrainsLogger.set_bypass_mode(True)
-    with TrainsLogger(output_uri=dirname) as trains_logger:
+    with pytest.warns(UserWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_trains_logger.py:610]: Here too

File: tests/ignite/contrib/handlers/test_trains_logger.py [hunk 13]
```diff
@@ -606,28 +607,29 @@ def test_integration_as_context_manager(dirname):
...
 
 def test_trains_disk_saver_integration():
     model = torch.nn.Module()
     to_save_serializable = {"model": model}
-
-    mock_logger = MagicMock(spec=TrainsLogger)
-    trains.Task.current_task = Mock(return_value=object())
-    trains_saver = TrainsSaver(mock_logger)
-    trains.binding.frameworks.WeightsFileHandler.create_output_model = MagicMock()
+    with pytest.warns(UserWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_trains_logger.py:628]: Here too

File: tests/ignite/contrib/handlers/test_trains_logger.py [hunk 14]
```diff
@@ -648,12 +650,11 @@ def test_trains_disk_saver_integration_no_logger():
     model = torch.nn.Module()
     to_save_serializable = {"model": model}
 
-    trains.Task.current_task = Mock(return_value=object())
-    trains.binding.frameworks.WeightsFileHandler.create_output_model = MagicMock()
-
-    trains_saver = TrainsSaver()
-
-    checkpoint = Checkpoint(to_save=to_save_serializable, save_handler=trains_saver, n_saved=1)
+    with pytest.warns(UserWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_trains_logger.py:653]: Here as well

File: tests/ignite/contrib/handlers/test_visdom_logger.py [hunk 15]
```diff
@@ -786,8 +786,9 @@ def _test(tag=None):
 
 def test_integration_no_server():
 
-    with pytest.raises(RuntimeError, match="Failed to connect to Visdom server"):
-        VisdomLogger()
+    with pytest.warns(PendingDeprecationWarning):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_visdom_logger.py:789]: Here as well

File: tests/ignite/contrib/engines/test_common.py (hunk 1 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/engines/test_common.py]: Done

File: tests/ignite/contrib/handlers/test_base_logger.py (hunk 2 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_base_logger.py:157]: They were accidentally indented, but it should be fixed now

File: tests/ignite/contrib/handlers/test_custom_events.py (hunk 3 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_custom_events.py]: Done

File: tests/ignite/contrib/handlers/test_custom_events.py (hunk 4 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_custom_events.py]: It seems it does, but that test looked like this before this PR

File: tests/ignite/contrib/handlers/test_custom_events.py (hunk 5 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_custom_events.py:27]: Done

File: tests/ignite/contrib/handlers/test_custom_events.py (hunk 6 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_custom_events.py]: Done

File: tests/ignite/contrib/handlers/test_custom_events.py (hunk 7 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_custom_events.py]: Done

File: tests/ignite/contrib/handlers/test_custom_events.py (hunk 8 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_custom_events.py]: Done

File: tests/ignite/contrib/handlers/test_lr_finder.py (hunk 9 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_lr_finder.py]: Done

File: tests/ignite/contrib/handlers/test_tqdm_logger.py (hunk 10 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_tqdm_logger.py]: Done

File: tests/ignite/contrib/handlers/test_trains_logger.py (hunk 11 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_trains_logger.py]: Done

File: tests/ignite/contrib/handlers/test_trains_logger.py (hunk 13 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_trains_logger.py]: Done

File: tests/ignite/contrib/handlers/test_trains_logger.py (hunk 14 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_trains_logger.py]: Done

File: tests/ignite/contrib/handlers/test_trains_logger.py (hunk 12 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_trains_logger.py]: Done

File: tests/ignite/contrib/handlers/test_visdom_logger.py (hunk 15 above)
Comment (InCogNiTo124) [reply tests/ignite/contrib/handlers/test_visdom_logger.py]: Done

**Comment by InCogNiTo124** [comment]
It seems sometimes some warnings are emitted, and other times they aren't, for example `tests/ignite/contrib/handlers/test_lr_finder.py:222`.

File: tests/ignite/contrib/handlers/test_custom_events.py [hunk 16]
```diff
@@ -8,55 +8,44 @@
...
-
-    with pytest.raises(ValueError):
-        CustomPeriodicEvent()
-
-    with pytest.raises(ValueError):
-        CustomPeriodicEvent(n_iterations=1, n_epochs=2)
+    with pytest.warns(DeprecationWarning, match=r"CustomPeriodicEvent is deprecated"):
+        with pytest.raises(ValueError):
+            CustomPeriodicEvent(n_iterations="a")
+            CustomPeriodicEvent(n_iterations=0)
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_custom_events.py:14]: This is incorrect. Control flow exits `with pytest.raises(ValueError):` on the first exception. Thus, other tests are not executed at all.

File: tests/ignite/contrib/handlers/test_custom_events.py [hunk 17]
```diff
@@ -8,55 +8,44 @@
...
-    with pytest.raises(ValueError):
-        CustomPeriodicEvent(n_epochs=10.0)
-
-    with pytest.raises(ValueError):
-        CustomPeriodicEvent()
-
-    with pytest.raises(ValueError):
-        CustomPeriodicEvent(n_iterations=1, n_epochs=2)
+    with pytest.warns(DeprecationWarning, match=r"CustomPeriodicEvent is deprecated"):
+        with pytest.raises(ValueError):
```
Comment (vfdev-5) [inline tests/ignite/contrib/handlers/test_custom_events.py:12]: Could you, please, add `match=r"..."` here as well. Thanks !

File: tests/ignite/contrib/handlers/test_custom_events.py (hunk 5 above)
Comment (vfdev-5) [reply tests/ignite/contrib/handlers/test_custom_events.py:27]: Seems not done, right ?

**Comment by InCogNiTo124** [comment]
`ignite/contrib/handlers/custom_events.py` has a typo in its warning message :D Since it's deprecated, it's not worth fixing it, is it?

**Comment by vfdev-5** [comment]
You can fix the typo :)

**Comment by vfdev-5** [comment]
@InCogNiTo124 my bad I haven't thought about that warnings can depend on python version. Case of `collection.abc` etc. Maybe we can ignore it as it is not related to our code.

**Comment by InCogNiTo124** [comment]
What I'm more worried about is that I can not deal with `ioloop.install()` warning from `test_setup_visdom_logging`. The warning is thrown somewhere deep in the visdom library, and, funnily enough, wherever I'd put `pytest.warns()` it would fail.

**Comment by vfdev-5** [comment]
> @vfdev-5 What I'm more worried about is that I can not deal with `ioloop.install()` warning from `test_setup_visdom_logging`. The warning is thrown somewhere deep in the visdom library, and, funnily enough, wherever I'd put `pytest.warns()` it would fail.

I think we can just ignore them as it is not our code is responsible for that warning. Let's remove those `pytest.warns()` that we can not handle and keep only those where we catch our `warning.warn`. What do you think ?

**Comment by InCogNiTo124** [comment]
That sounds doable

**Comment by InCogNiTo124** [comment]
I have removed the check for version dependant warnings, so I'd say this task is done.

**Comment by vfdev-5** [comment]
All warnings from the issue are solved ! 
But seems like I introduced another one with the recent commit. 
https://travis-ci.org/github/pytorch/ignite/jobs/694021111#L2120
Let me fix by myself, I'll commit it here and we can merge. 
Thanks again for working on this PR !

**Comment by InCogNiTo124** [comment]
Not a problem @vfdev-5, I'll come back for another one when I find the time :smile:

**Pull request merged**